activity.py

```diff
@@ -227,7 +242,9 @@ def write_file(self, file_path):
         data = {'seed': self.game.maze.seed,
                 'width': self.game.maze.width,
                 'height': self.game.maze.height,
-                'finish_time': self.game.finish_time}
+                'finish_time': self.game.finish_time,
+                'hole': self.game.hole}
```

Comment:
Three different terms for the same concept; self.game.hole is not "add_hole" or "add holes".

Reply:
I have corrected this.